### PR TITLE
Support DT_HASH

### PR DESCRIPTION
**Pull request by take-cheeze**

### CODE REVIEW

File: sold.cc [hunk 1]
```diff
@@ -107,6 +107,32 @@ uint32_t CalcGnuHash(const std::string& name) {
     return h;
 }
 
+uint32_t CalcHash(const std::string& name) {
+    uint32_t h = 0, g;
+    for (unsigned char c : name) {
+        h = (h << 4) + c;
+        if ((g = h & 0xf0000000)) {
```
Comment (shinh) [inline sold.cc:114]: I think we don't need this if-clause?

```
g = h & 0xf0000000;
h ^= g >> 24;
```

File: sold.cc [hunk 2]
```diff
@@ -107,6 +107,32 @@ uint32_t CalcGnuHash(const std::string& name) {
...
+            h ^= g >> 24;
+        }
+        h &= ~g;
```
Comment (shinh) [inline sold.cc:117]: `h ^= g;` would be simpler?

**Comment by take-cheeze** [comment]
:naruhodo: https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=bfd/elf.c;h=9ca42e10d8e5b28090d6ce73cc65fbd456210123;hb=HEAD#l195

**Comment by shinh** [comment]
LGTM, thanks again!

**Pull request merged**